adapters/bidmachine/bidmachine.go

```diff
@@ -0,0 +1,223 @@
+	errs := make([]error, 0, len(impressions))
+
+	for i, impression := range impressions {
+		if impression.Banner == nil && impression.Video == nil {
+			errs = append(errs, &errortypes.BadInput{
+				Message: "Bidmachine supports only banner or video ads",
+			})
+			continue
+		}
+
+		if impression.Banner != nil {
+			banner := impression.Banner
+			if banner.Format == nil {
+				errs = append(errs, &errortypes.BadInput{
+					Message: "banner format required",
+				})
+				continue
+			}
+			if len(banner.Format) == 0 {
+				errs = append(errs, &errortypes.BadInput{
+					Message: "banner format array is empty",
+				})
+				continue
+			}
+		}
+
+		if len(impression.Ext) == 0 {
+			errs = append(errs, errors.New("impression extensions required"))
+			continue
+		}
+		var bidderExt adapters.ExtImpBidder
+		err := json.Unmarshal(impression.Ext, &bidderExt)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		if len(bidderExt.Bidder) == 0 {
+			errs = append(errs, errors.New("bidder required"))
+			continue
+		}
+		var impressionExt openrtb_ext.ExtImpBidmachine
+		err = json.Unmarshal(bidderExt.Bidder, &impressionExt)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		url, err := a.buildEndpointURL(impressionExt)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		if impressionExt.SellerID == "" {
+			errs = append(errs, errors.New("Bidmachine seller_id is required"))
+			continue
+		}
+		if impressionExt.Path == "" {
+			errs = append(errs, errors.New("Bidmachine path is required"))
+			continue
+		}
+		if bidderExt.Prebid != nil && bidderExt.Prebid.IsRewardedInventory == 1 {
+			if impression.Banner != nil && !hasRewardedBattr(impression.Banner.BAttr) {
+				impression.Banner.BAttr = append(impression.Banner.BAttr, openrtb.CreativeAttribute(16))
+
+			}
+			if impression.Video != nil && !hasRewardedBattr(impression.Video.BAttr) {
+				impression.Video.BAttr = append(impression.Video.BAttr, openrtb.CreativeAttribute(16))
+			}
+		}
+		request.Imp = impressions[i : i+1]
+		body, err := json.Marshal(request)
+		if err != nil {
+			errs = append(errs, err)
+			continue
+		}
+		result = append(result, &adapters.RequestData{
+			Method:  "POST",
+			Uri:     url + "/" + impressionExt.Path + "/" + impressionExt.SellerID,
+			Body:    body,
+			Headers: headers,
+		})
+	}
+
+	request.Imp = impressions
+
+	if len(result) == 0 {
+		return nil, errs
+	}
+	return result, errs
+}
+
+func hasRewardedBattr(attr []openrtb.CreativeAttribute) bool {
+	for i := 0; i < len(attr); i++ {
+		if attr[i] == openrtb.CreativeAttribute(16) {
+			return true
+		}
+	}
+	return false
+}
+
+func (a *BidmachineAdapter) MakeBids(request *openrtb.BidRequest, _ *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	var errs []error
+
+	switch responseData.StatusCode {
+	case http.StatusNoContent:
+		fallthrough
+	case http.StatusServiceUnavailable:
+		return nil, nil
```

Comment:
Why does your bidding server normally respond with a 503 status code?

Reply:
Fixed, server is not supposed to respond with 503